src/tbtcv2-rewards/rewards.ts

```diff
@@ -0,0 +1,726 @@
+    }
+
+    // This is an example to illustrate a client's build version requirement.
+    // A client must be either on the second to latest version or latest.
+    //                       v1 or v2                v2
+    //           |-------------------------------\|-------|
+    // Timeline -|-------------*------------------*-------|->
+    //         Sep1            v2             v2+delay   Sep30
+    // Where:
+    // v2 was released Sep10
+    // delay = 14days
+    // v2 + delay = Sep10 + 14days = Sep24
+    // Between Sep1 - Sep24 a client is allowed to run v1 or v2
+    // Between Sep24 - Sep30 a client is allowed to run only v2
+
+    const buildVersion = instanceWithLatestBuildVersion.metric.version;
+    const latestClientRelease = clientReleases[0].split("_");
+    const latestClientTag = latestClientRelease[0];
+    const latestClientReleaseTimestamp = latestClientRelease[1];
+    if (clientReleases.length > 1) {
+      // A client is allowed to be on either of the two latest releases.
+      const secondToLatestClientRelease = clientReleases[1].split("_");
+      const secondToLatestClientTag = secondToLatestClientRelease[0];
+
+      let allowedDelayEndTimestamp =
+        latestClientReleaseTimestamp + ALLOWED_UPGRADE_DELAY;
+      if (allowedDelayEndTimestamp > endRewardsTimestamp) {
+        allowedDelayEndTimestamp = endRewardsTimestamp;
+      }
+
+      if (latestRegisteredBuildVersionTimestmap <= allowedDelayEndTimestamp) {
+        // A client's version can be on either 2 latest versions
+        requirements.set(
+          IS_VERSION_SATISFIED,
+          buildVersion.includes(latestClientTag) ||
+            buildVersion.includes(secondToLatestClientTag)
+        );
+      } else {
+        // The allowed delay for an upgrade is over. A client should be on the
+        // latest build version.
+        requirements.set(
+          IS_VERSION_SATISFIED,
+          buildVersion.includes(latestClientTag)
+        );
+      }
+    } else {
+      requirements.set(
+        IS_VERSION_SATISFIED,
+        buildVersion.includes(latestClientTag)
+      );
+    }
+
+    /// Start assembling peer data and weighted authorizations
+    operatorData[stakingProvider] = {
+      applications: Object.fromEntries(authorizations),
+      instances: convertToObject(instancesData),
+      requirements: Object.fromEntries(requirements),
+    };
+
+    if (
+      requirements.get(IS_BEACON_AUTHORIZED) &&
+      requirements.get(IS_TBTC_AUTHORIZED) &&
+      requirements.get(IS_UP_TIME_SATISFIED) &&
+      requirements.get(IS_PRE_PARAMS_SATISFIED) &&
+      requirements.get(IS_VERSION_SATISFIED)
+    ) {
+      const beacon = BigNumber.from(authorizations.get(BEACON_AUTHORIZATION));
+      const tbct = BigNumber.from(authorizations.get(TBTC_AUTHORIZATION));
+      let minApplicationAuthorization = beacon;
+      if (beacon.gt(tbct)) {
+        minApplicationAuthorization = tbct;
+      }
+
+      rewardData[stakingProvider] = {
```

Comment:
```suggestion
      rewardsData[stakingProvider] = {
```

Reply:
Sure, part of [c78a6b7](https://github.com/threshold-network/merkle-distribution/pull/35/commits/c78a6b730f5fc7045bf5fbbd34b65be39292a8f4)